### DISCUSSION TITLE
Improve selection options on task tries page

### DISCUSSION THREAD

**Discussion by otienoanyango** [apache/airflow#21691] [Ideas]
**Description**

Add a way of filtering the tasks you want graphed on the task tries UI page. At least an option to deselect all or select all would be quite welcome.

**Use case / motivation**

When there are many tasks, the task tries page does not allow filtering of what is shown in the graph which is useful when you want to only compare task duration for one or a few tasks. The current option is to deselect each of the tasks you don't want to see. Assuming 100 tasks, that's deselecting 95 if you're interested in only 5.

**Related Issues**
Not that I'm aware of

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by mik-laj** [comment]
This feature looks interesting. Would you like to working on it? This is an Open Source project and we don't have paid support. The fastest way to resolve a ticket is to contribute a change that will solve this problem.
If you are interested, here is the contributor's guide.
https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst

**Comment by otienoanyango** [comment]
I looked at the code and since nvd3 is used, it looks like it is not any straightforward to add  non-data buttons to the chart. I would have to look a little deeper if that's even a good solution.
